src/App.tsx

```diff
@@ -9,39 +9,42 @@ import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
 import { getUserPosts } from './api/posts';
-import { User } from './types/User';
-import { Post } from './types/Post';
 import { Counter } from './features/counter/Counter';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import { setError, setLoading, setPosts } from './features/postsSlice';
+import { removePost } from './features/selectedPostSlice';
 
 export const App: React.FC = () => {
-  const [posts, setPosts] = useState<Post[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
-
-  const [author, setAuthor] = useState<User | null>(null);
-  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const dispatch = useAppDispatch();
+  const {
+    items: posts,
+    loaded,
+    hasError,
+  } = useAppSelector(state => state.posts);
+  const { selectedUser } = useAppSelector(state => state.selectedUser);
+  const { selectedPost } = useAppSelector(state => state.currentPost);
```

Comment:
Create separate selector for each `useAppSelector` call

Read more: [link](https://redux.js.org/usage/deriving-data-selectors)